jbpm-wb-common/jbpm-wb-common-client/src/test/java/org/jbpm/workbench/common/client/list/AbstractMultiGridViewTest.java

```diff
@@ -232,4 +236,81 @@ public void testControlBulkOperationsDisabled() {
         getView().controlBulkOperations(extendedPagedTable);
         verify(mockButton).setEnabled(false);
     }
-}
\ No newline at end of file
+
+    @Test
+    public void testReloadColumnSortListWhenInitListTable() {
+        ListTable<T> listTable = mock(ListTable.class);
+        when(getView().getListGrid()).thenReturn(listTable);
+        GridPreferencesStore gridPreferencesStore = mock(GridPreferencesStore.class);
+        when(listTable.getGridPreferencesStore()).thenReturn(gridPreferencesStore);
+        GridSortedColumnPreference gridSortedColumnPreference = new GridSortedColumnPreference("test",
+                                                                                               true);
+        when(gridPreferencesStore.getGridSortedColumnPreference()).thenReturn(gridSortedColumnPreference);
+        ColumnSortList columnSortList = getColumnSortList("startDate",
+                                                          null);
+        when(listTable.getColumnSortList()).thenReturn(columnSortList);
+
+        ColumnMeta columnMeta = new ColumnMeta(getColumn("test"),
+                                               "test");
+        when(listTable.getColumnMetaList()).thenReturn(Arrays.asList(columnMeta));
+
+        assertEquals(1, columnSortList.size());
+
+        getView().reloadColumnSortList();
+
+        assertEquals("test", columnSortList.get(0).getColumn().getDataStoreName());
+        assertEquals("startDate", columnSortList.get(1).getColumn().getDataStoreName());
+        assertEquals(2, columnSortList.size());
```

Comment:
It is not clear to me why the size of columnsortlist has increased here, could you please help me understand?

Reply:
It have two cases for unit test, when columnSortList size less than 1, it mean init the grid table, and it will add gridSortedColumnPreference into columnSortList, if the size more than 1 , it mean it catch Column sorted event, do not add gridSortedColumnPreference into columnSortList.

Reply:
Could you please add a comment in code to explain where and why the size is increased? is it near getView().reloadColumnSortList(); ?

Reply:
ok